src/main/java/uk/gov/hmcts/reform/bulkscanning/services/SasTokenGeneratorService.java

```diff
@@ -0,0 +1,95 @@
+@Service
+public class SasTokenGeneratorService {
+    private final CloudBlobClient cloudBlobClient;
+    private final StorageCredentials storageCredentials;
+    private final AccessTokenConfiguration accessTokenConfiguration;
+
+    private static final Logger log = LoggerFactory.getLogger(SasTokenGeneratorService.class);
+
+    public SasTokenGeneratorService(
+        CloudBlobClient cloudBlobClient,
+        StorageCredentials storageCredentials,
+        AccessTokenConfiguration accessTokenConfiguration
+    ) {
+        this.cloudBlobClient = cloudBlobClient;
+        this.storageCredentials = storageCredentials;
+        this.accessTokenConfiguration = accessTokenConfiguration;
+    }
+
+    public String generateSasToken(String serviceName) {
+        //Based on the service name container reference would be returned for specific service
+        try {
+            log.info("SAS Token request received for service {} ", serviceName);
+
+            CloudBlobContainer cloudBlobContainer = cloudBlobClient.getContainerReference(serviceName);
+
+            return cloudBlobContainer.generateSharedAccessSignature(createSharedAccessPolicy(serviceName), null);
+        } catch (URISyntaxException | StorageException | InvalidKeyException e) {
+            throw new UnableToGenerateSasTokenException(e);
+        }
+    }
+
+    private SharedAccessBlobPolicy createSharedAccessPolicy(String serviceName) {
+        AccessTokenConfiguration.TokenConfig config = getTokenConfigForService(serviceName);
+
+        Calendar cal = new GregorianCalendar(TimeZone.getTimeZone("UTC"));
+        cal.setTime(new Date());
+        cal.add(Calendar.SECOND, config.getValidity());
+
+        SharedAccessBlobPolicy policy = new SharedAccessBlobPolicy();
+        policy.setPermissions(getBlobPermissions(config));
+        policy.setSharedAccessExpiryTime(cal.getTime());
+
+        return policy;
+    }
+
+    private AccessTokenConfiguration.TokenConfig getTokenConfigForService(String serviceName) {
+        return accessTokenConfiguration.getServiceConfig().stream()
+            .filter(tokenConfig -> tokenConfig.getServiceName().equalsIgnoreCase(serviceName))
+            .findFirst()
+            .orElseThrow(
+                () -> new ServiceConfigNotFoundException("No service configuration found for service " + serviceName)
+            );
+    }
+
+    private EnumSet<SharedAccessBlobPermissions> getBlobPermissions(AccessTokenConfiguration.TokenConfig config) {
+        EnumSet<SharedAccessBlobPermissions> blobPermissions = EnumSet.noneOf(SharedAccessBlobPermissions.class);
+
+        List<String> configuredPermissions = Arrays.asList(config.getPermissions().split(","));
+
+        configuredPermissions.forEach(perm -> {
+            if (EnumUtils.isValidEnum(SharedAccessBlobPermissions.class, perm)) {
+                blobPermissions.add(SharedAccessBlobPermissions.valueOf(perm));
+            }
+        });
+        return blobPermissions;
+    }
+}
```

Comment:
General note on the class: there are bunch of `AccessTokenConfiguration.TokenConfig` use cases. Can we just import the `TokenConfig` and use that instead?